File: gherkin/java/src/main/java/gherkin/ast/ScenarioDefinition.java
package gherkin.ast;

import java.util.Collections;
import java.util.List;

public abstract class ScenarioDefinition extends Node {
    private final String keyword;
    private final String name;
    private final String description;
    private final List<Step> steps;

    public ScenarioDefinition(Location location, String keyword, String name, String description, List<Step> steps) {
        super(location);
        this.keyword = keyword;
        this.name = name;
        this.description = description;
        this.steps = Collections.unmodifiableList(steps);
    }

    public String getName() {
        return name;
    }

    public String getKeyword() {
        return keyword;
    }

    public String getDescription() {
        return description;
    }

    public List<Step> getSteps() {
        return steps;
    }
}

File: gherkin/dotnet/Gherkin/Ast/ScenarioDefinition.cs
using System.Collections.Generic;

namespace Gherkin.Ast
{
    public abstract class ScenarioDefinition : IHasLocation, IHasDescription, IHasSteps
    {
        public Location Location { get; private set; }
        public string Keyword { get; private set; }
        public string Name { get; private set; }
        public string Description { get; private set; }
        public IEnumerable<Step> Steps { get; private set; }

        protected ScenarioDefinition(Location location, string keyword, string name, string description, Step[] steps)
        {
            Location = location;
            Keyword = keyword;
            Name = name;
            Description = description;
            Steps = steps;
        }
    }
}